cmd/nex/workload.go

```diff
@@ -208,6 +210,58 @@ func (s StopWorkload) Run(ctx context.Context, globals *Globals, w *Workload) er
 	return nil
 }
 
+type InfoWorkload struct {
+	WorkloadId   string `arg:"" description:"ID of the workload"`
+	WorkloadType string `name:"type" description:"Type of workload" default:"direct_start"`
+}
+
+func (InfoWorkload) AfterApply(globals *Globals) error {
+	return checkVer(globals)
+}
+
+func (i InfoWorkload) Validate() error {
+	var errs error
+
+	return errs
+}
+
+func (i InfoWorkload) Run(ctx context.Context, globals *Globals) error {
+	if globals.Check {
+		return printTable("Build Workload Configuration", append(globals.Table(), i.Table()...)...)
+	}
+
+	nc, err := configureNatsConnection(globals)
+	if err != nil {
+		return err
+	}
+
+	controller, err := nodecontrol.NewControlApiClient(nc, slog.New(slog.NewTextHandler(os.Stdin, nil)))
+	if err != nil {
+		return err
+	}
+
+	resp, err := controller.FindWorkload(i.WorkloadType, globals.Namespace, i.WorkloadId)
+	if errors.Is(err, nats.ErrTimeout) {
+		fmt.Println("Workload not found")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	w := columns.New("Information about Workload %s", i.WorkloadId)
+	w.AddRow("Name", resp.WorkloadSummary.Name)
+	w.AddRow("Start Time", resp.WorkloadSummary.StartTime)
+	w.AddRowIf("Run Time", resp.WorkloadSummary.Runtime, resp.WorkloadSummary.Runtime != "")
```

Comment:
haven't seen this `AddRowIf` before. Very nice.

Reply:
Stole it from nats.go 🤣